modules/extensions/src/data-filter/shader-module.ts

```diff
@@ -48,45 +58,48 @@ float dataFilter_reduceValue(vec3 value) {
 float dataFilter_reduceValue(vec4 value) {
   return min(min(value.x, value.y), min(value.z, value.w));
 }
-void dataFilter_setValue(DATAFILTER_TYPE valueFromMin, DATAFILTER_TYPE valueFromMax, DATACATEGORY_TYPE category) {
-  if (filter_enabled) {
+
+#ifdef DATAFILTER_ATTRIB
```

Comment:
The code is kind of hard to review as one has to guess what these `#defines` mean. Perhaps declaring a `defines` type above the shader with comments could be a way.

```
type defines = {
  /* ... */
  DATAFILTER_ATTRIB?: boolean,
 ...
};
```